File: mario.c
#include <stdio.h>
#include <cs50.h>

// Function Prototypes
void mario(int n);
void printSpaces(int h, int r);
void printHashes(int h, int r);

int main(void)
{
    // Prompt user for height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
   
    // Program ran successfuly
    mario(height);
    return 0;
}

void mario(int h)
{
    // Print every row of pyramid
    for (int rows = 0; rows < h; rows++)
    {
        printSpaces(h, rows);
        printHashes(h, rows);
        printf("  ");
        printHashes(h, rows);
        printf("\n");
    }
}

// Function to print spaces given height and row
void printSpaces(int h, int r)
{
    for (int spaces = h - 1; spaces > r; spaces--)
    {
        printf(" ");
    }
}

// Function to print hashes given height and row
void printHashes(int h, int r)
{
    for (int hashes = 0; hashes < r + 1; hashes++)
    {
        printf("#");
    }
}

File: mario.py
# Creates a mario half pyramid

from cs50 import get_int
from sys import exit


def main():
    # Prompts the user for an integer from 1 to 8
    while True:
        height = get_int("Height: ")
        if height >= 1 and height <= 8:
            break
    # Prints the half mario pyramid
    for i in range(height):
        printBlank(height - 1 - i)
        printHash(i + 1)
        printSpace()
        printHash(i + 1)
        print()
    # Program ran successfully
    exit(0)


# Prints the blank spaces
def printBlank(b):
    for i in range(b):
        print(" ", end="")


# Prints the hashes/blocks
def printHash(h):
    for i in range(h):
        print("#", end="")


# Prints a space between the half pyramids
def printSpace():
    print("  ", end="")


main()